File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let lines = [];
rl.on('line', input => {
    lines.push(input.trim());

    if (lines.length === parseInt(lines[0].split(' ')[1]) + 2) {
        rl.close()
    }
})

rl.on('close', () => {
    const [n, m] = lines.shift().split(' ').map(Number)
    const numList = lines.shift().split(' ').map(Number)
    let dp = new Array(n + 1).fill(0)

    for (let i = 1; i < n + 1; i++) {
        dp[i] = dp[i - 1] + numList[i - 1]
    } 

    for (let i = 0; i < m; i++) {
        const [start, end] = lines[i].split(' ').map(Number)
        console.log(dp[end] - dp[start - 1])
    }
})

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline

n, m = map(int, input().split())
nums = [0] + list(map(int, input().split()))

for i in range(1, n + 1):
    nums[i] += nums[i - 1]
    
for _ in range(m):
    start, end = map(int, input().split())
    print(nums[end] - nums[start - 1])